views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react'
+import { Box, Button } from '@mui/material'
+import { Add } from '@mui/icons-material'
+import { useOutletContext, Link } from 'react-router-dom'
+
+import ConfigurationsTable from '../../tables/ConfigurationsTable'
+import PageHeader from '../../components/PageHeader'
+import { UsersModel } from '../../models'
+import api from '../../api'
+import ShareConfigurationForm from '../../forms/ShareConfigurationForm'
+import { routes } from '../../routes/routes'
+
+const ConfigurationsPage = () => {
+  const {
+    configurations,
+    navigate,
+    setNotification,
+    setConfigurations,
+    setUser,
+    user,
+    users,
+  } = useOutletContext()
+  const { uid, preferences } = user
+  const [open, setOpen] = useState(false)
+  const [options, setOptions] = useState([])
+  const [formValues, setFormValues] = useState([])
+  const [currentConfigId, setCurrentConfigId] = useState('')
+
+  const handleNotification = (message) =>
+    setNotification({ open: true, message })
+  const onSubmit = async (data) => {
+    try {
+      const configAttributes = {
+        readers: UsersModel.createReadersList(data.readers),
+      }
+      await api.userConfigurations.update(
+        uid,
+        currentConfigId,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => {
+    setCurrentConfigId('')
+    setOpen(false)
+    setOptions([])
+    reset()
+  }
+  const handleFormModal = async (config) => {
+    const { owner, readers, _id } = config
+    const options = UsersModel.createUserFriendList(users, owner)
+    const values = UsersModel.createOptionsFromReaders(readers, owner)
+
+    setCurrentConfigId(_id)
+    setFormValues(values)
+    setOptions(options)
+    setOpen(true)
+  }
+  const handleDefaultConfiguration = async (configurationId) => {
+    try {
+      const userAttributes = {
+        ...user,
+        preferences: {
+          ...preferences,
+          config: preferences.config === configurationId ? '' : configurationId,
+        },
+      }
+      const updatedUser = await api.users.update(uid, userAttributes)
+
+      setUser(updatedUser)
+      handleNotification('Default Configuration Updated Successfully ')
```

Comment:
are these notifications part of the new design?

Reply:
These were part of the page before, but we can take it out

Reply:
[0241286](https://github.com/AMP-SCZ/dpdash/pull/487/commits/02412869aaf62d541a21ffa43e88c7647596c72d)